pyproject.toml

```diff
@@ -1,11 +1,53 @@
+[project]
+name = "dc_logging_utils"
+version = "2.0.0"
+description = "DC Logging client and log querying infrastructure"
+readme= "README.md"
+requires-python = "==3.12.*"
+dependencies = [
+    # Matches the Node version of the CDK in package.json
+    "aws-cdk-lib<3.0.0,>=2.195.0",
+    "aws_cdk.aws_lambda_python_alpha>=2.195.0a0",
+    "aws-cdk.aws-glue-alpha>=2.195.0a0",
+    "constructs>=10.0.0",
+    "cachetools==v5.2.0",
+    "wheel==0.38.4",
+    "urllib3==1.26.19",
+    "attrs==23.1.0",
+    "dc-logging-utils",
+]
+
+[dependency-groups]
+dev = [
+    "boto3==1.35.*",
+    "mypy-boto3-organizations",
+    "mypy-boto3-ssm",
+    "ruff==0.1.8",
+    "ipdb",
+    "pytest==7.2.2",
+    "pytest-ruff",
+    "pytest-cov",
+    "boto3-stubs[firehose,s3,sts,organizations,lambda]==1.28.80",
+    "psycopg[binary]",
+    "tqdm",
+    "moto[firehose,awslambda,apigateway,proxy,sts]==5.0.19",
+]
```

Comment:
Can we pin everything in both groups with `==` rather than ranges

Reply:
Done in [8457b38](https://github.com/DemocracyClub/dc_logging/pull/659/commits/8457b381c81ab591cefbc1c1013b15f26a9748e5).
I'm sure we'll get on to it in the package/application split, but just noting here that `dc_logging_client/log_entries.py` uses boto3, so we might want to figure out a way of making that accept a range.